qiskit/circuit/controlflow/if_else.py

```diff
@@ -0,0 +1,152 @@
+            tested for equality with a given ``int``, or as a tuple of a
+            ``Clbit`` to be compared to either a ``bool`` or an ``int``.
+        true_body: A program to be executed if ``condition`` evaluates
+            to true.
+        false_body: A optional program to be executed if ``condition``
+            evaluates to false.
+        label: An optional label for identifying the instruction.
+
+    If provided, ``false_body`` must be of the same ``num_qubits`` and
+    ``num_clbits`` as ``true_body``.
+
+    The classical bits used in ``condition`` must be a subset of those attached
+    to the circuit on which this ``IfElseOp`` will be appended.
+
+    **Circuit symbol:**
+
+    .. parsed-literal::
+
+             ┌───────────┐
+        q_0: ┤0          ├
+             │           │
+        q_1: ┤1          ├
+             │  if_else  │
+        q_2: ┤2          ├
+             │           │
+        c_0: ╡0          ╞
+             └───────────┘
+
+    """
+
+    def __init__(
+        self,
+        condition: Union[
+            Tuple[ClassicalRegister, int],
+            Tuple[Clbit, int],
+            Tuple[Clbit, bool],
+        ],
+        true_body: QuantumCircuit,
+        false_body: Optional[QuantumCircuit] = None,
+        label: Optional[str] = None,
+    ):
+        # Type checking generally left to @params.setter, but required here for
+        # finding num_qubits and num_clbits.
+        if not isinstance(true_body, QuantumCircuit):
+            raise CircuitError(
+                "IfElseOp expects a true_body parameter "
+                f"of type QuantumCircuit, but received {type(true_body)}."
+            )
+
+        num_qubits = true_body.num_qubits
+        num_clbits = true_body.num_clbits
+
+        super().__init__("if_else", num_qubits, num_clbits, [true_body, false_body], label=label)
+
+        try:
+            lhs, rhs = condition
+        except TypeError as err:
+            raise CircuitError(
+                "IfElseOp expects a condition argument as either a "
+                "Tuple[ClassicalRegister, int], a Tuple[Clbit, bool] or "
+                f"a Tuple[Clbit, int], but received {condition} of type "
+                f"{type(condition)}."
+            ) from err
+
+        if not (
+            (isinstance(lhs, ClassicalRegister) and isinstance(rhs, int))
+            or (isinstance(lhs, Clbit) and isinstance(rhs, (int, bool)))
+        ):
+            raise CircuitError(
+                "IfElseOp expects a condition argument as either a "
+                "Tuple[ClassicalRegister, int], a Tuple[Clbit, bool] or "
+                f"a Tuple[Clbit, int], but receieved a {type(condition)}"
+                f"[{type(lhs)}, {type(rhs)}]."
+            )
+
+        self.condition = condition
+
+    @property
+    def params(self):
+        return self._params
+
+    @params.setter
+    def params(self, parameters):
+        true_body, false_body = parameters
+
```

Comment:
This is a good question. I disagree slightly that `instruction.params = true_body, None` would be unintuitive, because at this point, your instruction is an `IfElseOp` so there's an expectation of needing to explicitly specify both branches (unlike say `QuantumCircuit.if_test` vs `QuantumCircuit.if_else`). The question of param specific getters and setters I think is best for #7087 .

Reply:
That's fair.  I wouldn't say that individual parameter setters depends on #7087 if a subclass wants to give them names (i.e. have `IfElseOp.true_body` is a `property`), but I also don't really care - I'm happy to make mutating things difficult, because I don't like mutation!